### PR TITLE
AV-36571 #comment Added file service resource in Terraform.

### PR DESCRIPTION
**Pull request by gitshrikant**
AV-36571 #comment Added file service resource in Terraform.

### CODE REVIEW

File: avi/utils.go [hunk 1]
```diff
@@ -411,3 +411,29 @@ func ApiDeleteSystemDefaultCheck(d *schema.ResourceData) bool {
...
+	file_path := d.Get("file_path").(string)
+	var err error
+	if http_method == "POST" {
+		//For multipart file upload
+		err := client.AviSession.PostMultipartRequest(http_method, uri, file_path)
+		if err != nil {
+			log.Printf("[ERROR] MultipartUploadOrDownload Error uploading file %v %v", file_path, err)
+			return err
+		}
+	}
```
Comment (grastogi23) [inline avi/utils.go]: let us use the Switch premitive in Go. That is more appropriate than the if then else here.
https://tour.golang.org/flowcontrol/9

File: avi/resource_avi_fileservice.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			Required: true,
+			ForceNew: true,
+		},
+	}
+}
+
+func resourceAviFileService() *schema.Resource {
+	return &schema.Resource{
+		Read:   ResourceAviFileServiceRead,
+		Create: ResourceAviFileServiceUploadOrDownload,
```
Comment (grastogi23) [inline avi/resource_avi_fileservice.go]: why no update?

File: avi/resource_avi_fileservice.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+import (
+	"github.com/avinetworks/sdk/go/clients"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+)
+
+func ResourceFileServiceSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"http_method": &schema.Schema{
```
Comment (grastogi23) [inline avi/resource_avi_fileservice.go]: File service should not have http_method. It should simply be uri and file_path and also file content.

**Comment by gitshrikant** [comment]
Before merging this PR, We need to merge all SDK PR's. This PR take reference of latest changes made to SDK master branch. As those changes haven't merged in SDK that's why Travis CI fails for this PR.

**Pull request merged**